src/URI.js

```diff
@@ -575,6 +575,10 @@
       string = '/' + string;
     }
 
+    if (parts.port) {
```

Comment:
Ok, I've removed the positive testcase, improved the port validation code and 3 negative tests in total.